### PR TITLE
onednn: add version 2.5.2.

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot maintainers

**Review by adamjstewart — APPROVED** [review]
It looks like there are new options available for `cpu_runtime` and `gpu_runtime`: https://github.com/oneapi-src/oneDNN#runtime-dependencies

You can add these if you want, or I can merge this PR as is if you don't care about those options.

**Comment by tkameyama** [comment]
I don't care those option, now.
Please merge the PR.

**Pull request merged**
